pkg/trace/agent/agent.go

```diff
@@ -189,50 +183,73 @@ func (a *Agent) Process(p *api.Payload) {
 	ss := new(writer.SampledSpans)
 	var envtraces []stats.EnvTrace
 	a.PrioritySampler.CountClientDroppedP0s(p.ClientDroppedP0s)
-	for _, t := range p.Traces {
-		if len(t) == 0 {
+
+	for i := 0; i < len(p.TracerPayload.Chunks); {
+		t := p.TracerPayload.Chunks[i]
```

Comment:
```suggestion
		chunk := p.TracerPayload.Chunks[i]
```
I think this will make the rest of the code more understandable, within the new context. What do you think?